CommonTools/UtilAlgos/interface/ProductFromFwdPtrProducer.h

```diff
@@ -25,7 +25,7 @@ namespace edm {
 
 
   template < class T, class H = ProductFromFwdPtrFactory<T> >
-  class ProductFromFwdPtrProducer : public edm::EDProducer {
+    class ProductFromFwdPtrProducer : public edm::global::EDProducer<> {
```

Comment:
Are you confident that every factory that exists or ever will exist don't use global variables?